sdk/inc/azure/core/az_log.h

```diff
@@ -76,11 +76,20 @@ typedef void (*az_log_message_fn)(az_log_classification classification, az_span
  * @brief Allows the application to specify which #az_log_classification types it is interested in
  * receiving.
  *
- * @details If no classifications are set (`NULL`), the application will receive log messages for
- * all #az_log_classification values.
+ * @param[in] classifications __[nullable]__ An array of #az_log_classification values, terminated
+ * by #AZ_LOG_END_OF_LIST.
  *
- * @param[in] classifications __[nullable]__ An array of az_log_classification values, terminated by
- * #AZ_LOG_END_OF_LIST.
+ * @details If no classifications are set (\p classifications is `NULL`), the application will
+ * receive log messages for all #az_log_classification values.
+ * @details If \p classifications is not `NULL`, it must point to an array of
+ * #az_log_classification, terminated by #AZ_LOG_END_OF_LIST.
+ * @details In constrast to \p classifications being `NULL`, \p classifications pointing to an empty
+ * array (which still should be terminated by #AZ_LOG_END_OF_LIST), states that an aplication is not
+ * interested in receiving any message, regardless of classification.
+ *
+ * @warning Users should not change the \p classifications array elements, once it is passed to this
+ * function. If \p classifications array is allocated on a stack, program behavior in multi-threaded
+ * environment is undefined.
```

Comment:
We should add code snippets, and maybe even some tests for such edge cases, but not critical.

Reply:
Thank you Ahson, I just added update to samples to use safer pattern of declaring classifications for samples and for readme.

I so far don't know if it is possible to update the test to verify the conditions that we are documenting against, given that when that happens, the timing needs to be just precisely right; plus they are going to verify that the behavior is going to be undefined as we are telling, I am not sure what an error or success for such test case would mean.

As of the multi-threaded testing, I think eventually we should get some sort of a mechanism for testing that allows us to run multiple threads, and do so in such a controlled precision (with stopping and going), that they are guaranteed to reproduce a race condition; maybe we don't really need multiple threads for that, and single thread simulating two threads, by jumping from one location to another at precisely pre-programmed moments is better, but this is a bigger item that won't be fixed within this PR, let's revisit the whole idea as part of a bigger picture, there could be some other places (the ones that we wrote with multi-threading in mind) where we would use such mechanism to verify that the multi threaded behavior is exactly as we intended it to be, with 100% accuracy.